source/common/common/utility.h

```diff
@@ -709,4 +709,32 @@ class InlineString : public InlineStorage {
   char data_[];
 };
 
+/**
+ * Utilities for working with containers.
+ */
+class Containers {
+public:
+  /**
+   * Remove elements that match the predicate from the list by swapping them to
+   * the end and truncating. Iterates over c once, order is not preserved.
+   *
+   * @param c The container on which to operate
+   * @param predicate A function which will be called once for every entry in c,
+   * which returns true if that entry should be removed.
+   */
+  template <class Container, typename T>
+  static void removeMatchingElements(Container& c, std::function<bool(const T&)> predicate) {
```

Comment:
Can you try std::function(T) here? If there are cases where this needs to be a const ref, then the caller can explicitly say that in the template qualification.

Reply:
I meant: `std::function<bool(T)>`